### PR TITLE
[TIMOB-23436]:ListView Control's Height not setting in iOS

### PR DESCRIPTION
**Pull request by vijaysingh-axway**
https://jira.appcelerator.org/browse/TIMOB-23436

### CODE REVIEW

File: iphone/Classes/TiUIListView.m [hunk 1]
```diff
@@ -1762,7 +1762,7 @@ - (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPa
     }
     if (TiDimensionIsDip(height)) {
         return height.value;
-    } else if (TiDimensionIsAutoSize(height)) {
+    } else if (TiDimensionIsAutoSize(height) || TiDimensionIsUndefined(height)) {
```
Comment (hansemannn) [inline iphone/Classes/TiUIListView.m]: The test-case works for me. But I am wondering if we shouldn't include this as well:
```objc
    } else if (TiDimensionIsAuto(height) || TiDimensionIsAutoSize(height) || TiDimensionIsUndefined(height)) {
```
And I also noticed that the template do has a height and width specified, so it shouldn't go until that line, can you check that before approving? Thx! Good work!

Comment (vijaysingh-axway) [inline iphone/Classes/TiUIListView.m]: While setting Ti.UI.auto for row height , it gives undefined. Rather I found that for Ti.UI.Fill check similar condition should applied . Updated check with TiDimensionIsAutoFill(height). 
Thanks.

Comment (hansemannn) [inline iphone/Classes/TiUIListView.m]: Mhh, I'd be really curious if checking for size and fill is a good idea. Pretty sure it breaks something else.

Comment (vijaysingh-axway) [inline iphone/Classes/TiUIListView.m]: I can not think of any use case where it can break something. Are you ?

Comment (vijaysingh-axway) [inline iphone/Classes/TiUIListView.m]: @hansemannn Ping.

**Comment by hansemannn** [comment]
Please add more details why exactly this change fixes this issue. Also, add UI tests for all layout-specific use-cases for all if-statements available in the source. I am trying to avoid a major layout regression like in https://jira.appcelerator.org/browse/TIMOB-24753, so we better check this more detailled.

**Review by hansemannn — APPROVED** [review]
LGTM!

**Review by ssjsamir — APPROVED** [review]
**FR Passed:** Was able to set the ListView Control's height without any issues. 

**Test Steps**
+ Downloaded the SDK Build from this PR
+ Created a new Titanium project
+ Copied the test case from the comments in https://jira.appcelerator.org/browse/TIMOB-23436
+ Ran the application 
+ Was able to see the ListView height change 
+ Changed the ListView height in the code 
+ Was able to see the new height

**Test Environment**
Appcelerator Command-Line Interface, version 6.2.2
iPhone 6S (10.0)
iPhone 6S Simulator(10.2)
Operating System Name: Mac OS X El Capitan
Operating System Version: 10.11.6
Node.js Version: 6.10.1
Xcode: 8.2
Appcelerator Studio: 4.9.0.201705251638

**Pull request merged**